scripts/update-submodule.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Update python-base submodule and collect release notes.
+# Usage:
+#   scripts/update-submodule.sh
+#
+# Create a commit "generated python-base update" and send a PR to the main repo.
+# NOTE: not all python-base changes are user-facing, and you may want to remove
+# some of the non-user-facing release notes from CHANGELOG.md.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+repo_root="$(git rev-parse --show-toplevel)"
+declare -r repo_root
+cd "${repo_root}"
+
+source scripts/util/changelog.sh
+
+# TODO(roycaihw): make the script send a PR by default (standalone mode), and
+# have an option to make the script reusable by other release automations.
+TARGET_RELEASE=${TARGET_RELEASE:-"v$(grep "^CLIENT_VERSION = \"" scripts/constants.py | sed "s/CLIENT_VERSION = \"//g" | sed "s/\"//g")"}
+SUBMODULE_SECTION=${SUBMODULE_SECTION:-"\*\*Submodule Change:\*\*"}
+
+# update submodule
+git submodule update --remote
+pulls=$(git diff --submodule | egrep -o "^  > Merge pull request #[0-9]+" | sed 's/  > Merge pull request #//g')
+
+# download release notes
+release_notes=""
+submodule_repo="kubernetes-client/python-base"
+echo "Downloading release notes from submodule repo."
+for pull in ${pulls}; do
+  pull_url="https://github.com/$submodule_repo/pull/${pull}"
+  echo "+++ Downloading python-base patch to /tmp/${pull}.patch"
+  curl -o "/tmp/${pull}.patch" -sSL "${pull_url}.patch"
+  subject=$(grep -m 1 "^Subject" "/tmp/${pull}.patch" | sed -e 's/Subject: \[PATCH//g' | sed 's/.*] //')
+  pull_uid="$submodule_repo#${pull}"
+  release_note="- ${subject} [${pull_uid}](${pull_url})\n"
```

Comment:
again, release note is from subject here, better to have a release note section in PR

Reply:
done